src/vsr/replica.zig

```diff
@@ -5626,7 +5626,8 @@ pub fn ReplicaType(
                     .view_change => if (header.view == self.view) return,
                     else => unreachable,
                 },
-                .recovering_head => {},
+                // We need a start_view from any other replica — don't request it from ourselves.
+                .recovering_head => if (self.primary_index(header.view) == self.replica) return,
```

Comment:
I am wondering if we can assert this situation more narrowly. We got a message for a view where we are the primary, and we are recovoring from a crash. I think it follows that `header.view == self.view or (header.command == .start_view_change or header.command == .do_view_change)`. 

Basically, double-checking that, after we restart, we correctly remember our last view. If we restart and notice a message from a view where we were a primary, something's very wrong! Though, that assert probably belongs to `ignore_prepare_ok` rather than here.

Reply:
https://github.com/tigerbeetledb/tigerbeetle/pull/455

Reply:
> If we restart and notice a message from a view where we were a primary, something's very wrong!

Would this always be wrong? For example, what if a message were misdirected? I think it would still be wrong, because we don't expect a primary to author the message. Just double-checking...